code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm

```diff
@@ -288,46 +288,63 @@
 	var/mob/living/carbon/Xenomorph/Queen/X = owner
 	if(!X.check_state())
 		return
-	if(X.observed_xeno)
-		var/mob/living/carbon/Xenomorph/T = X.observed_xeno
 
-		if(T.banished)
-			to_chat(X, SPAN_XENOWARNING("This xenomorph is already banished!"))
-			return
+	var/choice = tgui_input_list(X, "Choose a xenomorph to banish:", "Banish", X.hive.totalXenos)
 
-		if(T.hivenumber != X.hivenumber)
-			to_chat(X, SPAN_XENOWARNING("This xenomorph doesn't belong to your hive!"))
-			return
+	if(!choice)
+		return
 
-		// No banishing critted xenos
-		if(T.health < 0)
-			to_chat(X, SPAN_XENOWARNING("What's the point? They're already about to die."))
-			return
+	var/mob/living/carbon/Xenomorph/T
 
-		var/confirm = alert(X, "Are you sure you want to banish [T] from the hive? This should only be done with good reason.", , "Yes", "No")
-		if(confirm == "No")
-			return
+	for(var/mob/living/carbon/Xenomorph/xeno in X.hive.totalXenos)
+		if(html_encode(xeno.name) == html_encode(choice))
+			T = xeno
+			break
 
-		var/reason = stripped_input(X, "Provide a reason for banishing [T]. This will be announced to the entire hive!")
-		if(isnull(reason))
-			to_chat(X, SPAN_XENOWARNING("You must provide a reason for banishing [T]."))
-			return
+	if(T == X)
+		to_chat(X, SPAN_XENOWARNING("You cannot banish yourself."))
+		return
 
-		if(!X.check_state() || !X.check_plasma(plasma_cost) || X.observed_xeno != T || T.health < 0)
-			return
+	if(T.banished)
+		to_chat(X, SPAN_XENOWARNING("This xenomorph is already banished!"))
+		return
 
-		// Let everyone know they were banished
-		xeno_announcement("By [X]'s will, [T] has been banished from the hive!\n\n[reason]", X.hivenumber, title=SPAN_ANNOUNCEMENT_HEADER_BLUE("Banishment"))
-		to_chat(T, FONT_SIZE_LARGE(SPAN_XENOWARNING("The [X] has banished you from the hive! Other xenomorphs may now attack you freely, but your link to the hivemind remains, preventing you from harming other sisters.")))
+	if(T.hivenumber != X.hivenumber)
+		to_chat(X, SPAN_XENOWARNING("This xenomorph doesn't belong to your hive!"))
+		return
 
-		T.banished = TRUE
-		T.hud_update_banished()
-		T.lock_evolve = TRUE
+	// No banishing critted xenos
+	if(T.health < 0)
+		to_chat(X, SPAN_XENOWARNING("What's the point? They're already about to die."))
+		return
 
-		message_staff("[key_name_admin(X)] has banished [key_name_admin(T)]. Reason: [reason]")
+	var/confirm = alert(X, "Are you sure you want to banish [T] from the hive? This should only be done with good reason.", , "Yes", "No")
```

Comment:
You should mention it that this carries an OOC power too as that they won't be able to rejoin as xeno for half an hour and, therefore, abusing this will have serious punishment and should not be used without a serious reason. Changelog will quickly move and new players won't know about this feature.